### PR TITLE
vagrant and ansible first stab at using galaxy

### PR DESCRIPTION
**Pull request by vccabral**
DO NOT MERGE - JUST REVIEW 

@kave @kurtw @rosskarchner @marcesher 

Please see PR. This is a starter example of utilizing ansible's role's path and ansible galaxy in an attempt to minimize code duplication across our projects as well as follow and define a few new best practices.

Note 'ansible/vars/db.yml' contains some plain text passwords that will be vault-ified in a later PR but they are placeholders for now.

instructions - after installing ansible
```
$ ansible-galaxy install -r ansible/requirements.yml
$ vagrant up
```

Note that 2 playbooks are taken from the most popular open source repos.
Note that 1 playbook is taken from a popular open source repo and customized for our needs (and a PR was placed against the main open source repo for elasticsearch)
Note that our requirements for those roles are described in a versioned requirements.yml file instead of being included inside the project directory.
Note that those roles are downloaded from github, github enterprises, etc, and placed into your local roles path (for me it is /Users/cabralv/homebrew/bin/ansible/roles) but this is configurable. 

Obviously there is a ton of work to do to get this ready for even local vagrant but part of this is asking for your ideas to contribute to the way we create ansible playbooks as opposed to how we are creating this playbook. Feel free to give feedback and thanks!

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  config.vm.provider "virtualbox" do |vb|
+    vb.memory = "1024"
+  end
+
+  config.vm.provision "ansible" do |ansible|
+      ansible.playbook = "ansible/cfgov_refresh_setup.yml"
+      ansible.verbose = 'v'
+      ansible.groups = {
+          "db" => ["default"],
```
Comment (vccabral) [inline Vagrantfile]: Vagrant configuration gives the 1 local box all the roles of the major components of the entire app stack.

File: ansible/cfgov_refresh_setup.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+- hosts: db
```
Comment (vccabral) [inline ansible/cfgov_refresh_setup.yml]: Each major component of app stack is broken into it's main components.

File: ansible/cfgov_refresh_setup.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    - { role: geerlingguy.mysql }
+
+- hosts: app
+  sudo: yes
+  sudo_user: root
+  vars_files:
+    - vars/app.yml
+  roles:
+    - { role: geerlingguy.java }
+    - { role: vccabral.elasticsearch }
```
Comment (vccabral) [inline ansible/cfgov_refresh_setup.yml]: points to my customized third party elasticsearch (https://github.com/vccabral/ansible-role-elasticsearch/tree/add_version)

File: ansible/vars/db.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+mysql_root_password: super-secure-password
```
Comment (vccabral) [inline ansible/vars/db.yml]: Should be in vault or other KMS like credstash

File: ansible/vars/app.yml [hunk 5]
```diff
@@ -0,0 +1 @@
+__elasticsearch_version: "1.7.x"
```
Comment (vccabral) [inline ansible/vars/app.yml]: Latest is 2.x but we need 1.7.x

**Comment by kave** [comment]
cool looks good. i forgot to mention we'll also need [pdfReactor](http://download.realobjects.de/PDFreactor_6_3_6828_3_unix_installer.tar.gz) installed within the box. location doesn't matter we currently have it installed in the `/opt/` location.

you'll also need to add it to the python path `export PYTHONPATH="${PYTHONPATH}:/opt/PDFreactor/wrappers/python/lib"`

start server: `/opt/PDFreactor/bin/pdfreactorwebservice start`

_tip_ : i ran into an issue being able to connect the server within unitybox incase you have the same issue you'll need to:

- Edit the PDFReactor/jetty/etc/jetty.xml to use the host name 0.0.0.0.

There are some license global vars to consider but we talk about that offline

**Comment by vccabral** [comment]
@kave thanks for the feedback. I have PDFreactor -> zip written down in my notes so I'm glad you reminded me so we can include that in this config.

**Comment by vccabral** [comment]
https://github.cfpb.gov/ConsumerResponse/complaint-analytics-search-tool-box/tree/master/provisioners/roles/elasticsearch @mehtad wrote a pretty great elasticsearch role that, on first inspection, seems better than the open source playbook I stole.

**Comment by rosskarchner** [comment]
@vccabral it seems to be trying to write to /etc on my mac?

```
- downloading role from https://github.com/geerlingguy/ansible-role-mysql/archive/1.9.1.tar.gz
- extracting geerlingguy.mysql to /etc/ansible/roles/geerlingguy.mysql
- error: you do not have permission to modify files in /etc/ansible/roles/geerlingguy.mysql
- geerlingguy.mysql was NOT installed successfully.
```

**Comment by vccabral** [comment]
@rosskarchner it writes to your ansible roles_path directory. Mine defaulted to my brew install directory but you can customize it a few ways.

http://docs.ansible.com/ansible/intro_configuration.html#roles-path

**Comment by rosskarchner** [comment]
Aha-- that's probably a difference of note, for installing via homebrew or pip.

**Comment by vccabral** [comment]
@kave & @kurtw I'd love to sit down with one of you for a few minutes to go over extra details of getting PDFReactor working.

**Comment by vccabral** [comment]
@kave can you please run a test locally?

```
$ git clone git@github.com:vccabral/cfgov-refresh.git
$ cd cfgov-refresh
$ ansible-galaxy install -r ansible/requirements.yml
$ vagrant up
```

After you get the box up, use the setup scripts and then we can dry run the site to make sure it is working.

**Comment by vccabral** [comment]
@rosskarchner any thoughts  on the initial design?

**Comment by rosskarchner** [comment]
@vccabral I just tried this out, and id died at:

```
TASK: [Add environment variables to bashrc] *********************************** 
changed: [default] => {"backup": "", "changed": true, "msg": "line added"}

TASK: [Migrate django db to match application state] ************************** 
fatal: [default] => failed to transfer file to /home/vagrant/.ansible/tmp/ansible-tmp-1456340115.46-34486355913313/command:

Received message too long 1449751156


FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/Users/karchnerr/cfgov_refresh_setup.retry

default                    : ok=65   changed=45   unreachable=1    failed=0   

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
```

**Comment by vccabral** [comment]
@rosskarchner please remove any echo in the .env script to solve your error.

**Comment by vickumar1981** [comment]
@kave can you close this pr?

it's addressed as part of : https://github.com/cfpb/cfgov-refresh/pull/1534

**Pull request closed without merging**